include/util/guidance/name_announcements.hpp

```diff
@@ -26,7 +26,8 @@ namespace guidance
 // Name Change Logic
 // Used both during Extraction as well as during Post-Processing
 
-inline std::string longest_common_substring(const std::string &lhs, const std::string &rhs)
+template <typename StringView>
+inline StringView longest_common_substring(const StringView &lhs, const StringView &rhs)
```

Comment:
Why do we need a template type for the string view here?

Reply:
it is not needed. Changed to the explicit type `utill::StringView`